apps/emqx_mongodb/src/emqx_mongodb.erl

```diff
@@ -97,7 +97,15 @@ fields(sharded) ->
     ] ++ mongo_fields();
 fields(topology) ->
     [
-        {pool_size, fun emqx_connector_schema_lib:pool_size/1},
+        {pool_size,
+            hoconsc:mk(
+                pos_integer(),
+                #{
+                    deprecated => {since, "5.1.1"},
+                    importance => ?IMPORTANCE_HIDDEN,
+                    converter => fun(_, _) -> 1 end
```

Comment:
Nit. IMO we don't have to go to such great lengths, default = 1 should be enough? Given the option is both deprecated and hidden. If we want to disregard it completely, isn't it better to set to 1 somewhere here?

https://github.com/emqx/emqx/blob/38bd8a81852caeda71ec58e874785c9736c49ef0/apps/emqx_mongodb/src/emqx_mongodb.erl#L370-L371

Also, does this converter work fine with `make_serializable`? Probably yes, but generally such converter seems "risky".

Reply:
We also want to override any previously user-set values.

Hmmm I'll have to look into the `make_serializable` thing and where it's used.  But I assume it works, as we already do [something similar](https://github.com/emqx/emqx/blob/1ea92d400925fa5732868ec270071d680e7420ea/apps/emqx_bridge_mongodb/src/emqx_bridge_mongodb.erl#L54) in MongoDB bridge.

Reply:
👍🏼 Got it. I have a feeling we need to work out some approach to how to deal with such deprecations.

Overall, for me it sounds more straightforward to just have `init_topology_options(R, [{pool_size, 1} | Acc]);` and tell a user though a log message that their configuration option is not being respected and why.

Reply:
I've moved the override to the `on_start` callback and added a log to warn about the override.